src/bvals/cc/bvals_cc_in_one.cpp

```diff
@@ -421,18 +556,23 @@ void SendAndNotify(MeshData<Real> *md) {
 //          Guarantees that buffers for MeshBlocks on the same rank are done, but MPI
 //          communication between ranks may still be in process.
 
-// TODO(pgrete) should probaly be moved to the bvals or interface folders
+// TODO(pgrete) should probably be moved to the bvals or interface folders
 TaskStatus SendBoundaryBuffers(std::shared_ptr<MeshData<Real>> &md) {
   Kokkos::Profiling::pushRegion("Task_SendBoundaryBuffers_MeshData");
 
-  auto boundary_info = md->GetSendBuffers();
-  bool cache_is_valid = boundary_info.is_allocated();
+  for (int b = 0; b < md->NumBlocks(); ++b) {
+    md->GetBlockData(b)->SetLocalNeighborAllcoated();
+  }
 
-  auto buffers_used = ResetSendBuffers(md.get(), cache_is_valid);
+  auto boundary_info = md->GetSendBuffers();
+  auto sending_nonzero_flags = md->GetSendingNonzeroFlags();
+  bool have_cache = boundary_info.is_allocated();
+  auto buffers_used = ResetSendBuffers(md.get(), have_cache);
 
-  if (!cache_is_valid) {
+  if (!have_cache || (buffers_used != boundary_info.size())) {
```

Comment:
Excellent question. I've improved this now to actually cache the allocation status in `MeshData` and it will recreate the buffers if the allocation status of any variable has changed.